### PR TITLE
[DomCrawler] fix HTML5 parser integration

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Spotted while reviewing #30892
The current logic is context-dependent: by changing the order of calls, you can get different behaviors.

### CODE REVIEW

**Review by jakzal — APPROVED** [review]

**Comment by stof** [comment]
I don't understand which issue you are trying to solve here ?

We cannot add content multiple time (as all nodes have to belong to the same document), and calling `->html()` before adding content does not make sense and will not reach this code anyway.

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -190,13 +191,7 @@ public function addContent($content, $type = null)
...
     {
         // Use HTML5 parser if the content is HTML5 and the library is available
-        if (!$this->html5Parser
-            && class_exists(HTML5::class)
-            && '<!doctype html>' === strtolower(substr(ltrim($content), 0, 15))) {
-            $this->html5Parser = new HTML5(['disable_html_ns' => true]);
-        }
-
-        $dom = null !== $this->html5Parser ? $this->parseHtml5($content, $charset) : $this->parseXhtml($content, $charset);
+        $dom = null !== $this->html5Parser && strspn($content, " \t\r\n") === stripos($content, '<!doctype html>') ? $this->parseHtml5($content, $charset) : $this->parseXhtml($content, $charset);
```
Comment (stof) [inline src/Symfony/Component/DomCrawler/Crawler.php]: this is not left-trimming the content anymore to find the doctype. Is it expected ?

Comment (nicolas-grekas) [reply src/Symfony/Component/DomCrawler/Crawler.php]: yes it is, see strspn

**Comment by nicolas-grekas** [comment]
> I don't understand which issue you are trying to solve here.

The code makes little either: one should not have to know how the overall class works to understand local methods. I'm making the code have a better locality so one can reason about it in a simpler way.

**Pull request merged**